float32 x          
float32 y          
float32 velocity    
float32 target_x  
float32 target_y   
